### PR TITLE
examples: Add multiple-binaries

### PR DESCRIPTION
**Pull request by Patryk27**
Closes https://github.com/nix-community/naersk/issues/272.

### CODE REVIEW

File: readme.nix [hunk 1]
```diff
@@ -15,25 +15,37 @@ let
...
+
+      params = builtins.readFile (
+        pkgs.runCommand "docparse"
+          { buildInputs = [ docparse ]; }
+          "docparse ${./config.nix} > $out"
+      );
+
+    in
+    pkgs.writeText "readme" (
+      builtins.replaceStrings
```
Comment (Patryk27) [inline readme.nix:42]: The sed-based approach didn't work on MacOS:
```
; ./script/gen
Updating README
this derivation will be built:
  /nix/store/pfvinxyrj3x27wrmvlcw0zbzkwlfn2kn-readme-body.drv
building '/nix/store/pfvinxyrj3x27wrmvlcw0zbzkwlfn2kn-readme-body.drv'...
sed: couldn't open temporary file /nix/store/sedbfv96n: Operation not permitted
error: builder for '/nix/store/pfvinxyrj3x27wrmvlcw0zbzkwlfn2kn-readme-body.drv' failed with exit code 4;
       last 1 log lines:
       > sed: couldn't open temporary file /nix/store/sedbfv96n: Operation not permitted
       For full logs, run 'nix log /nix/store/pfvinxyrj3x27wrmvlcw0zbzkwlfn2kn-readme-body.drv'.
```
... so I figured that switching to an (almost) pure-Nix approach will be better.

File: README.md [hunk 2]
```diff
@@ -226,6 +228,11 @@ process, rest is passed-through into `mkDerivation`.
 | `postInstall` | Optional hook to run after the compilation is done; inside this script, `$out/bin` contains compiled Rust binaries. Useful if your application needs e.g. custom environment variables, in which case you can simply run `wrapProgram $out/bin/your-app-name` in here. Default: `false` |
 | `usePureFromTOML` | Whether to use the `fromTOML` built-in or not. When set to `false` the python package `remarshal` is used instead (in a derivation) and the JSON output is read with `builtins.fromJSON`. This is a workaround for old versions of Nix. May be used safely from Nix 2.3 onwards where all bugs in `builtins.fromTOML` seem to have been fixed. Default: `true` |
 
+
+## Examples
+
+See: <./examples>.
```
Comment (nmattia) [inline README.md:234]: link's broken for me!

File: readme.nix [hunk 3]
```diff
@@ -15,25 +15,37 @@ let
...
-        in
-        p' == "Cargo.lock" || p' == "Cargo.toml" || p' == "src" || p' == "src/main.rs"
-    ) ./docparse;
+    src = builtins.filterSource
+      (
+        p: t:
+          let
+            p' = pkgs.lib.removePrefix (toString ./docparse + "/") p;
+          in
+          p' == "Cargo.lock" || p' == "Cargo.toml" || p' == "src" || p' == "src/main.rs"
```
Comment (nmattia) [inline readme.nix:24]: was this just reformatted?

File: readme.nix (hunk 1 above)
Comment (nmattia) [reply readme.nix:42]: This has the unfortunate side effect of importing from derivation!

Comment (Patryk27) [reply readme.nix:42]: Yeah, but that's just run locally, right? Naersk itself remains fully IFD-free 😇

File: README.md (hunk 2 above)
Comment (Patryk27) [reply README.md]: whoops, fixed!

File: readme.nix (hunk 1 above)
Comment (nmattia) [reply readme.nix:42]: heh alright

**Review by nmattia — APPROVED** [review]

File: readme.nix (hunk 1 above)
Comment (nmattia) [reply readme.nix:42]: btw that sed error looks odd, I think it's just a matter of changing the `sed` slightly on macos

**Pull request merged**